server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -401,6 +401,24 @@ public boolean equalsAliases(Metadata other) {
         return indexMapBuilder.build();
     }
 
+    /**
+     * Finds the parent data streams, if any, for the specified concrete indices.
+     */
+    public ImmutableOpenMap<String, IndexAbstraction.DataStream> findDataStreams(String[] concreteIndices) {
```

Comment:
I wonder if this should not be a private method on `TransportGetIndexAction` instead? Seems it mainly translates from the lookup to the map structure.